modules/drivers/redshift/test/metabase/driver/redshift_test.clj

```diff
@@ -149,3 +157,38 @@
                :parameters [{:type :date/all-options
                              :target [:dimension [:template-tag "date"]]
                              :value "past30years"}]}))))))
+
+(deftest redshift-types-test
+  (mt/test-driver
+    :redshift
+    (testing "Redshift specific types should be synced correctly"
+      (let [db-details   (tx/dbdef->connection-details :redshift)
+            tbl-nm       "redshift_specific_types"
+            qual-tbl-nm  (str rstest/session-schema-name "." tbl-nm)
+            view-nm      "late_binding_view"
+            qual-view-nm (str rstest/session-schema-name "." view-nm)]
+        (mt/with-temp
+          Database
+          [database {:engine :redshift, :details db-details}]
+          ;; create a table with a CHARACTER VARYING and a NUMERIC column, and a late bound view that selects from it
+          (#'rstest/execute!
+            (str "DROP TABLE IF EXISTS %1$s;%n"
+                 "CREATE TABLE %1$s(weird_varchar CHARACTER VARYING(50), numeric_col NUMERIC(10,2));%n"
+                 "CREATE OR REPLACE VIEW %2$s AS SELECT * FROM %1$s WITH NO SCHEMA BINDING;")
+            qual-tbl-nm
+            qual-view-nm)
+          ;; sync the schema again to pick up the new view (and table, though we aren't checking that)
+          (sync/sync-database! database)
+          (is
+            (contains?
+              (db/select-field :name Table :db_id (u/get-id database)) ; the new view should have been synced
+              view-nm))
+          (let [table-id (db/select-one-id Table :db_id (u/get-id database), :name view-nm)]
+            (is
+              ;; and its columns' :base_type should have been identified correctly
+              (=
+                [{:name "weird_varchar", :database_type "character varying(50)", :base_type :type/Text}
+                 {:name "numeric_col", :database_type "numeric(10,2)", :base_type :type/Decimal}]
+                (map
+                  (partial into {})
+                  (db/select [Field :name :database_type :base_type] :table_id table-id))))))))))
```

Comment:
Both `is` and `map` here should be https://guide.clojure.style/#one-space-indent

Reply:
Done.